src/core/controllers/oidc.go

```diff
@@ -244,6 +248,98 @@ func (oc *OIDCController) Callback() {
 	}
 }
 
+func (oc *OIDCController) RedirectLogout() {
+	sessionData := oc.GetSession(tokenKey)
+	ctx := oc.Ctx.Request.Context()
+	if sessionData == nil {
+		log.Error("OIDC session token not found.")
+		oc.SendInternalServerError(fmt.Errorf("OIDC session token not found"))
+		return
+	}
+	if err := oc.DestroySession(); err != nil {
+		log.Errorf("Error occurred in LogOut: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	oidcSettings, err := config.OIDCSetting(ctx)
+	if err != nil {
+		log.Errorf("Failed to get OIDC settings: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	if oidcSettings == nil {
+		log.Error("OIDC settings is missing.")
+		oc.SendInternalServerError(fmt.Errorf("OIDC settings is missing"))
+		return
+	}
+	if !oidcSettings.Logout {
+		oc.Controller.Redirect("/", http.StatusFound)
+		return
+	}
+	tk, ok := sessionData.([]byte)
+	if !ok {
+		log.Error("Invalid OIDC session data format.")
+		oc.SendInternalServerError(fmt.Errorf("invalid OIDC session data format"))
+		return
+	}
+	token := oidc.Token{}
+	if err := json.Unmarshal(tk, &token); err != nil {
+		log.Errorf("Error occurred in Unmarshal: %v", err)
+		oc.SendInternalServerError(err)
+		return
+	}
+	if oidcSettings.LogoutOffline && token.RefreshToken != "" {
+		sessionType, err := getSessionType(token.RefreshToken)
+		if err == nil {
+			// If the session is offline, revoke the refresh token
+			if strings.ToLower(sessionType) == "offline" {
+				if oidc.EndpointsClaims.RevokeURL != "" {
+					if err := revokeOIDCRefreshToken(oidc.EndpointsClaims.RevokeURL, token.RefreshToken, oidcSettings.ClientID, oidcSettings.ClientSecret); err != nil {
+						log.Errorf("Failed to revoke the offline session: %v", err)
+						oc.SendInternalServerError(err)
+						return
+					}
+				} else {
+					log.Warning("Unable to logout OIDC offline session since the 'revoke_endpoint' is not set.")
+				}
+			}
+		} else {
+			log.Warningf("Invalid refresh token for offline session: %s, error: %v", token.RefreshToken, err)
+		}
+	}
+
+	if token.RawIDToken == "" {
+		log.Warning("Unable to log out user session due to the empty ID token.")
+		oc.Controller.Redirect("/", http.StatusFound)
+		return
+	}
+	if _, err := oidc.VerifyToken(ctx, token.RawIDToken); err != nil {
```

Comment:
I don't think we need to verify the token here, we can just redirect the user to the `EndSessionURL`.  

Verifying the token here may cause negative impact, it's possible the ID token is expired at this moment and this should not block user from logging out from Harbor's perspective..

Reply:
removed.